packages/common-scripts/tests/locktime_pool.test.ts

```diff
@@ -435,3 +449,402 @@ test("Don't update capacity directly when deduct", async (t) => {
   t.deepEqual(inputCapacitiesBefore, inputCapacitiesAfter);
   t.deepEqual(outputCapacitiesBefore, outputCapacitiesAfter);
 });
+
+const depositHeader = {
+  compact_target: "0x20010000",
+  dao: "0x8eedf002d7c88852433518952edc28002dd416364532c50800d096d05aac0200",
+  epoch: "0xa000500283a",
+  hash: "0x41d081cd95d705c4e80a6b473f71050efc4a0a0057ee8cab98c4933ad11f0719",
+  nonce: "0x98e10e0a992f7274c7dc0c62e9d42f02",
+  number: "0x19249",
+  parent_hash:
+    "0xd4f3e8725de77aedadcf15755c0f6cdd00bc8d4a971e251385b59ce8215a5d70",
+  proposals_hash:
+    "0x0000000000000000000000000000000000000000000000000000000000000000",
+  timestamp: "0x17293289266",
+  transactions_root:
+    "0x9294a800ec389d1b0d9e7c570c249da260a44cc2790bd4aa250f3d5c83eb8cde",
+  uncles_hash:
+    "0x0000000000000000000000000000000000000000000000000000000000000000",
+  version: "0x0",
+};
+const withdrawHeader = {
+  compact_target: "0x20010000",
+  dao: "0x39d32247d33f90523d37dae613dd280037e9cc1d7b01c708003d8849d8ac0200",
+  epoch: "0xa0008002842",
+  hash: "0x156ecda80550b6664e5d745b6277c0ae56009681389dcc8f1565d815633ae906",
+  nonce: "0x7ffb49f45f12f2b30ac45586ecf13de2",
+  number: "0x1929c",
+  parent_hash:
+    "0xfe601308a34f1faf68906d2338e60246674ed1f1fbbad3d8471daca21a11cdf7",
+  proposals_hash:
+    "0x0000000000000000000000000000000000000000000000000000000000000000",
+  timestamp: "0x1729cdd69c9",
+  transactions_root:
+    "0x467d72af12af6cb122985f9838bfc47073bba30cc37a4075aef54b0f0768f384",
+  uncles_hash:
+    "0x0000000000000000000000000000000000000000000000000000000000000000",
+  version: "0x0",
+};
+
+class RpcMocker {
+  constructor() {}
+
+  async get_header(hash: string) {
+    if (hash === depositHeader.hash) {
+      return depositHeader;
+    }
+    if (hash === withdrawHeader.hash) {
+      return withdrawHeader;
+    }
+
+    throw new Error(`Error header hash!`);
+  }
+
+  async get_transaction(hash: string): Promise<any> {
+    if (hash === bobSecpDaoWithdrawInput.out_point!.tx_hash) {
+      return {
+        tx_status: {
+          block_hash: bobSecpDaoWithdrawInput.block_hash,
+        },
+        transaction: {
+          inputs: [
+            {
+              previous_output: bobSecpDaoDepositInput.out_point,
+            },
+          ],
+        },
+      };
+    }
+
+    if (hash === bobSecpDaoDepositInput.out_point!.tx_hash) {
+      return {
+        tx_status: {
+          block_hash: bobSecpDaoDepositInput.block_hash,
+        },
+      };
+    }
+
+    throw new Error("Error transaction hash!");
+  }
+}
+
+function cloneObject<T>(obj: T): T {
```

Comment:
It's better to use [lodash's cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep) method than the `JSON.parse/stringify` workaround for efficiency and safety reasons, eg. `JSON.parse/stringify` would cause data lost for certain properties  like function, Dates,  `undefined`(exists in the test data), etc.

Reply:
Replaced with `Immutable.fromJS(obj).toJS()` .

Reply:
Reverted to using `JSON.parse/stringify`, it's faster than `Immutable.fromJS(obj).toJS()`.
The main purpose for deep clone is cloning nested object, will not include functions or Dates, and key of undefined value lost is unimportant.